exec_test.go

```diff
@@ -247,30 +263,233 @@ func TestCommandStartWait(t *testing.T) {
 
 			cmd := tc.distro.Command(ctx, tc.cmd)
 
-			// BEFORE_START block
+			// BeforeStart block
 			if tc.cancelOn == BeforeStart {
 				cancel()
 			}
+			var stdout, stderr *bufio.Reader
+			if tc.stdoutPipe {
+				pr, err := cmd.StdoutPipe()
+				require.NoError(t, err, "Unexpected failure in call to (*Cmd).StdoutPipe")
+				stdout = bufio.NewReader(pr)
+
+				_, err = cmd.StdoutPipe()
+				require.Error(t, err, "Unexpected success calling (*Cmd).StdoutPipe twice")
+			}
+			if tc.stderrPipe {
+				pr, err := cmd.StderrPipe()
+				require.NoError(t, err, "Unexpected failure in call to (*Cmd).StderrPipe")
+				stderr = bufio.NewReader(pr)
+
+				_, err = cmd.StderrPipe()
+				require.Error(t, err, "Unexpected success calling (*Cmd).StderrPipe twice")
+			}
+
+			err := cmd.Wait()
+			require.Error(t, err, "Unexpected success calling (*Cmd).Wait before (*Cmd).Start")
 
 			cmd.Stdin = 0
-			cmd.Stderr = 0
-			cmd.Stdout = 0
-			err := cmd.Start()
+			err = cmd.Start()
 
-			// AFTER_START block
+			// AfterStart block
 			if tc.cancelOn == AfterStart {
 				cancel()
 			}
 			if requireErrors(t, tc, AfterStart, err) {
 				return
 			}
 
+			if stdout != nil {
+				text, err := stdout.ReadString('\n')
+				if err != nil {
+					require.ErrorIs(t, err, io.EOF, "Unexpected failure reading from StdoutPipe")
+				}
+				assert.Equal(t, tc.wantStdout, text, "Mismatch in piped stdout")
+			}
+
+			if stderr != nil {
+				text, err := stderr.ReadString('\n')
+				if err != nil {
+					require.ErrorIs(t, err, io.EOF, "Unexpected failure reading from StderrPipe")
+				}
+				assert.Equal(t, tc.wantStderr, text, "Mismatch in piped stderr")
+			}
+
+			err = cmd.Start()
+			require.Error(t, err, "Unexpected success calling (*Cmd).Start twice")
+
+			_, err = cmd.StdoutPipe()
+			require.Error(t, err, "Unexpected success calling (*Cmd).StdoutPipe after (*Cmd).Start")
+
+			_, err = cmd.StderrPipe()
+			require.Error(t, err, "Unexpected success calling (*Cmd).StderrPipe after (*Cmd).Start")
+
 			err = cmd.Wait()
 
-			// AFTER_WAIT block
+			// AfterWait block
 			if requireErrors(t, tc, AfterWait, err) {
 				return
 			}
+
+			err = cmd.Wait()
+			require.Error(t, err, "Unexpected success calling (*Cmd).Wait twice")
+		})
+	}
+}
+
+func TestCommandOutPipes(t *testing.T) {
+	d := newTestDistro(t, jammyRootFs)
+
+	testCases := map[string]struct {
+		cmd    string
+		stdout bool
+		stderr bool
+
+		wantRead string
```

Comment:
Similar to above, "want" would be enough, but if you prefer `wantRead` because basically this is a side effect (what was written in the buffer and not a function call output directly), I’m fine with it, tell me what you prefer! At least, this is prefixed by `want` :)

Reply:
I like it either way, let's stick to `want`. It is true that it is a side effect, but it is the relevant result of that function's happy path, so it makes sense for it to be `want`.